### ISSUE TITLE
Invalid JSON output

### ISSUE THREAD

**Issue by blanxii** [TALP-UPC/FreeLing#49]
Hello!!

First of all thanks for this amazing job!

I have an issue when I try to run as a JSON output format, seems to be invalid. Could you check it?

How to reproduce:

```
echo "Martin McGuinness, ex viceprimer ministro de Irlanda del Norte y y antiguo comandante del Ejército Republicano Irlandés (IRA), ha fallecido a los 66 años. Fue una figura clave en el proceso de paz en el país y hace sólo dos meses que se apartó de la vida política. Es con profundo pesar y tristeza que hemos sabido de la muerte de nuestro amigo y camarada Martin McGuinness, que falleció en la madrugada en Derry (Irlanda del Norte). Se le echará mucho de menos por todos los que le conocieron”, indicó la nota del partido republicano Sinn Fein en un comunicado. El republicano irlandés ha muerto a consecuencia de una rara enfermedad cardiaca. Martin McGuinness ha fallecido en el hospital de Derry Altnagelvin rodeado de su familia, según informa The Guardian." | analyze -f /usr/local/Cellar/freeling/4.0_4/share/freeling/config/es.cfg --nec --output json
```

Copy the ouput and check with any validator, for instance: http://json.parser.online.fr/

Thanks!

**Comment by lluisp** [comment]
I just fixed it in master.  You can pull the changes
Alternatively, you can just wrap the output you get in square brackets " [ ... ] " to get valid json

Thanks!

**Comment by blanxii** [comment]
Thanks for the answer @lluisp :)

The problem stills in master, again, if I test with a another text...

```echo "A guest post by Shannon Appelcline, a versatile author and programmer who currently works as the lead iOS developer for Skotos Tech, an online entertainment company. In the past two years, he’s written seven iOS games for them, all based on tabletop releases by popular German designers. The first of these was Reiner Knizia’s Money (2010)—which has also been ported to MacOS—while the most recent was Reiner Knizia’s Modern Art: The Card Game (2011). Shannon’s two most recent books show the breadth of his interests. They are iOS4 in Action (2011), published by Manning Publications, and Designers & Dragons: A History of the Roleplaying Game Industry (2011), published by Mongoose Publishing.

In Getting Started with Homebrew, I talked about what the Homebrew package management system is, why you’d want to use it, and how to install it and access its basic functionality.

This post expands upon those basic ideas by talking about how to keep your Homebrew up to date — and so ensure it’s doing its job optimally." | analyze -f /usr/local/Cellar/freeling/4.0_4/share/freeling/config/es.cfg --nec --output json``` 

Could you please check it back?

Thanks!

**Comment by blanxii** [comment]
Got it, if the input text have new lines, it breaks the JSON :)

**Comment by lluisp** [comment]
No, it is not the newlines.
The problem is that if there are more than one sentence, you get a list of json objects:
 {sentence1 analysis}, {sentence2 analysis}
but this list is not wrapped in "[" "]" as it should be:
[ {sentence1 analysis}, {sentence2 analysis} ]

So, if you just wrap the output in [...], it will work.
In any case, it is fixed now in master

**Comment by arademaker** [comment]
You can also see the bug as a feature: http://jsonlines.org

**Comment by arademaker** [comment]
Group of sentences that end in the same line ? I didn't understand it ? Do you mean paragraphs ?

**Comment by lluisp** [comment]
> You can also see the bug as a feature: http://jsonlines.org

Good to know...  However the output is not a single line per object...

> Group of sentences that end in the same line ? I didn't understand it ? Do you mean paragraphs?

If the input comes from a text processor, each line will probably contain a paragraph. But if it comes from other source, there may be line breaks in the middle of sentences.

In `--mode corpus` (the default), the  `analyze` program outputs a sentence analysi as soon as the sentence is completed.
So, if there is a line with several sentences, it will output all of them toghether.
If a sentence starts in one line but continues and ends in the next one, it will be output when the second line is read and the sentence completed (thoghether with other sentences that the second line may contain).

So, basically, all sentences that are completed in the same line (ie. end there) are output toghether
